assets/js/src/admin.js

```diff
@@ -26,3 +26,120 @@
 		} );
 	} );
 } )( window, jQuery );
+
+(function (window, $) {
+	'use strict';
+
+	var RSADisablePlugin = {
+
+		els: {
+			dialog: document.getElementById( 'rsa-disable-dialog' ),
+			placeholderA: document.getElementById( 'rsa-operator-a' ),
+			placeholderB: document.getElementById( 'rsa-operator-b' )
+		},
+
+		variables: {
+			operatorA: 0,
+			operatorB: 0,
+			expectedAnswer: 0,
+			disablingURL: null
+		},
+
+		openDialog: function ( event ) {
+			event.preventDefault();
+			$( this.els.dialog ).dialog( 'open' );
+		},
+
+		getRandomNum: function () {
+			return Math.floor( Math.random() * 10 );
+		},
+
+		isExpectedAnswer: function () {
+			var userResult = parseInt( document.getElementById( 'rsa-user-result' ).value, 10 );
+
+			if ( userResult === this.variables.expectedAnswer ) {
+				return true;
+			}
+
+			return false;
+		},
+		dialogSettings: function () {
+			var self = this;
+
+			$( this.els.dialog ).dialog({
+				title: '⚠️ ' + rsaAdmin.strings.warning,
+				dialogClass: 'wp-dialog',
+				autoOpen: false,
+				draggable: false,
+				width: 'auto',
+				modal: true,
+				resizable: false,
+				buttons: [
+					{
+						text: rsaAdmin.strings.confirm,
+						click: function () {
+							$( this ).dialog( 'close' );
```

Comment:
closing immediately is optimistic :) lets at least add an error message if the process fails

Reply:
The log method is more as a nice to have feature, were a log option value is updated every time this plugin is disabled network-wide. The actual disabling action is handled by WP core, with the JS redirect.